src/nccl_ofi_sendrecv.c

```diff
@@ -2074,20 +2101,34 @@ static int get_ep(nccl_net_ofi_device_t *base_dev,
 	 * initialize endpoint if necessary. */
 	nccl_net_ofi_sendrecv_ep_t *ep = pthread_getspecific(device->ep_key);
 	if (!ep) {
+
 		/* Allocate endpoint */
 		ep = calloc(1, sizeof(nccl_net_ofi_sendrecv_ep_t));
 		if (!ep) {
 			ret = -ENOMEM;
 			NCCL_OFI_TRACE(NCCL_INIT | NCCL_NET,
-				       "Unable to allocate sendrecv endpoint");
+			"Unable to allocate sendrecv endpoint");
 			goto unlock;
 		}
+#if HAVE_NEURON
+		if (!ep->domain) {
```

Comment:
Why do you need this if? You just calloc'd the endpoint, so `ep->domain` should always be `NULL` here.

Reply:
ack.